ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java

```diff
@@ -0,0 +1,455 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.invoker;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import io.ballerina.projects.BuildOptions;
+import io.ballerina.projects.BuildOptionsBuilder;
+import io.ballerina.projects.DiagnosticResult;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.JBallerinaBackend;
+import io.ballerina.projects.JarResolver;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.directory.SingleFileProject;
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.api.values.BFuture;
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.scheduling.Strand;
+import io.ballerina.shell.Diagnostic;
+import io.ballerina.shell.DiagnosticReporter;
+import io.ballerina.shell.exceptions.InvokerException;
+import io.ballerina.shell.snippet.Snippet;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+
+/**
+ * Invoker that invokes a command to evaluate a list of snippets.
+ * <p>
+ * State of an invoker persists all the information required.
+ * {@code reset} function will clear the invoker state.
+ * <p>
+ * Context of an invoker is the context that will be used to
+ * fill the template. This should be a logic-less as much as possible.
+ * Invoker and its context may be tightly coupled.
+ *
+ * @since 2.0.0
+ */
+public abstract class ShellSnippetsInvoker extends DiagnosticReporter {
+    public static final String MODULE_NOT_FOUND_CODE = "BCE2003";
+    // TODO: After configurables can be supported, change this to that file location
```

Comment:
Shall we link the issue created here? Along with the todo?

Reply:
#28662 Created an issue

Reply:
I mean shall we add it to the comment in the code along with the todo?